Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferSetProxy.cpp

```diff
@@ -215,11 +224,34 @@ void RemoteImageBufferSetProxy::didPrepareForDisplay(ImageBufferSetPrepareBuffer
 
         m_pendingFlush->setHandles(WTFMove(handle));
     }
-    if (m_remoteRenderingBackendProxy)
-        m_remoteRenderingBackendProxy->didPrepareForDisplay(*this);
+
+    m_prepareForDisplayIsPending = false;
+    if (m_shutdownIsPending) {
+        callOnMainRunLoop([strongThis = Ref { *this }]() {
+            assertIsMainRunLoop();
+            Locker locker { strongThis->m_lock };
+            strongThis->doShutdown();
```

Comment:
I don't have a good suggestion except holding streamconnection reference here and removing the workqueue messagereceiver in the work queue task. That should work, but it creates a ref cycle RIBSP->streamconnection->connection->workqueumessagehandler(ribsp)

Reply:
I thought that'd mostly be ok, since we'd still remove the workQueueMessageReceiver from the `didClose` notification. The second remove from the new connection would be invalid, but harmless.

I think the cycle part is ok, as long as we always get an explicit close, and always get either the message we expect deliver or a `didClose` notification to break them.

I wasn't sure if I could call `removeWorkQueueMessageReceiver` from the work queue, but that indeed simplifies things.